### ISSUE TITLE
tinyxml

### ISSUE THREAD

**Issue by jakoch** [fifengine/fifengine#903]
Hey,

because i arrived at building the TinyXML dependency in the appveyor VS toolchain
i use my chance to raise some questions about it's current state:

Fifengine ships TinyXML v1 - v2.5.3 internally
The latest version of TinyXML v1 is [v2.6.2](https://sourceforge.net/projects/tinyxml/files/tinyxml/2.6.2/)
TinyXML v1 is deprecated and unmaintained (latest release 2011-05-15).
It's time to move...

- [ ] the dependency is not listed in the wiki
   - please update https://github.com/fifengine/fifengine/wiki/dependencies and add TinyXML v1 there
   - please include a note that this is TinyXML v1 and not TinyXML v2, which is another product line

- [ ] remove the `engine/core/ext/tinyxml` folder?
  - Question: are you ok, if i remove the interally shipped version of TinyXML?
   - i don't see any significant changes, when compared to the external lib
   - that means, we stop shipping it internally and pull the dependency always externally

- [ ]  upgrade TinyXML v1 to either TinyXML v2 or RapidXML or PugiXML
  - [ ] which one? 
  - [ ] please create a new standalone issue/ticket to track the upgrade process, when you are ready to work on this

Regards, Jens

**Comment by prock-fife** [comment]
@jakoch very well done.  @vtchill is probably more familiar with this as I believe he was the one who add the TinyXML dependency originally.  We have had discussions in the past to remove it from EXT and we have always landed on keeping it there.  I'm just going to throw this out there and suggest we remove it and upgrade to TinyXML v2.  I cant think of a particular reason not to do so.

**Comment by vtchill** [comment]
Agree this is very well done.

I also agree that moving to tinyXML v2 makes sense. I'm not sure on the level of effort but I think the API has a good amount of change. I think it was only included in ext so that we didn't have to build yet another dependency in the dev kit and have a static library to link to. I am fine with removing and pulling external.

**Comment by LinuxDonald** [comment]
@jakoch what about to use RapidXML? because its already used in fifengine-atlas-creator https://github.com/fifengine/fifengine-atlas-creator/tree/master/src/rapidxml

**Comment by jakoch** [comment]
Yes, we/i could also replace the dependency in the atlas-creator. I prefer to solve the appveyor based Qt build issue firstly, so that it builds and is green, before we switch the dependencies. I'm in favor of using PugiXml, because it's fast and has a nice API.
